gson/src/main/java/com/google/gson/internal/bind/ReflectiveTypeAdapterFactory.java

```diff
@@ -213,9 +232,12 @@ private Adapter(ObjectConstructor<T> constructor, Map<String, BoundField> boundF
 
       out.beginObject();
       try {
-        for (BoundField boundField : boundFields.values()) {
-          if (boundField.writeField(value)) {
-            out.name(boundField.name);
+        for (Map.Entry<String, BoundField> entry : boundFields.entrySet()) {
+          String name = entry.getKey();
+          BoundField boundField = entry.getValue();
+          String defaultName = boundField.name;
+          if (name.equals(defaultName) && boundField.writeField(value)) {
```

Comment:
Wouldn't it be simpler to just create a boundField that has `serialized = false` for each of the alternates? Also a lot less string comparison in the default encoding loop.